HT5.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+import random
+
+
+class Publication:
+    def __init__(self, title):
+        self.title = title
+
+    def get_publication_text(self):
+        body = input('Type text: ')
+        return body
+
+    def write_to_list(self, publication_list):
+        publication_list.insert(0, pub.title)
+        publication_list.insert(1, self.get_publication_text())
+        return publication_list
+
+    def write_to_file(self, publicution_list):
+        file = open('myfile.txt', 'a+')
+        file.write('---------------------------------' + '\n')
+        file.write(f'{publicution_list[0]}\n{publicution_list[1]}\n{publicution_list[2]}\n')
+        file.write('---------------------------------' + '\n\n')
+
+
+class News(Publication):
+    def __init__(self, title, publish_datetime):
+        Publication.__init__(self, title=title)
+        self.publish_datetime = publish_datetime
+
+    def get_publish_date(self):
+        return self.publish_datetime.strftime('%Y-%m-%d %H:%M')
+
+    def get_news_location(self):
+        location = input('Type city name: ')
+        return location
+
+    def write_to_list_news(self, publication_list):
+        publication_list.append(f'{inp_location}, {self.get_publish_date()}')

```

Comment:
Same notice like in previous comment, please take news location as function argument, not global variable

Reply:
changed